### DISCUSSION TITLE
Get some time informations about a request like curl/libcurl

### DISCUSSION THREAD

**Discussion by git001** [hyperium/hyper#3441] [Ideas]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
I have created a simple c-program to measure some http(s) timings. It's nothing fancy or rocket science but it makes the job to measure the https request time.
https://github.com/git001/c-program/blob/main/libcurl-time.c

I use the libcurl's [curl_easy_getinfo](https://curl.se/libcurl/c/curl_easy_getinfo.html) function for that with one of the parameters `CURLINFO_TOTAL_TIME_T`.

Is there something similar in hyper or planned to add?

**Describe alternatives you've considered**
I have used the curl cli with the [write-out](https://curl.se/docs/manpage.html#-w) parameter which is quite handy for network debugging. It would be nice to have something similar in rust.

```
echo |curl -sSko /dev/null --max-time 6 \
  -w "%{http_code} namelookup %{time_namelookup} connect %{time_connect} pretransfer %{time_pretransfer} starttransfer %{time_starttransfer} appconnect %{time_appconnect} total %{time_total}\n" \
  https://172.xxx.xxxx.xxx/
```

**Additional context**
No additional context.

**Comment by seanmonstar** [comment]
There isn't anything built in to hyper that exposes the timings. At least not yet. Adding support is desired, but the design is not yet known. A likely way would be if `tracing` stabilized, and then `hyper` documented some events that could be listened for (https://github.com/hyperium/hyper/issues/2678).

**Reply by git001** [reply]
Thanks @seanmonstar for the fast answer. I will keep an eye on `tracing`
